### DISCUSSION TITLE
Work around issues in load_from_checkpoint() using trainer.predict()

### DISCUSSION THREAD

**Discussion by TimoSommer** [Lightning-AI/pytorch-lightning#13466] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hey all,

I recently noticed that `.load_from_checkpoint()` has some unintuitive features, for example not setting the correct `model.current_epoch` (#12819) and needing users to manually save the hyperparameters. 

I also noticed that if I just use the `self.trainer.predict` api, then `self.trainer.model` is automatically loaded with the correct weights and  `current_epoch` is also correctly the epoch of the checkpoint I saved, in contrast to when using `.load_from_checkpoint()`, plus the original, correct hyperparamters:
```
y_pred = self.trainer.predict(
                                    model=self.trainer.model,
                                    dataloaders=dataloader,
                                    ckpt_path='best'
                                    )
``` 
I am not sure though, if this also correctly loads optimizer states and everything else. 

I am wondering if this is a valid workaround or if some parts are not loaded this way. I guess there will be some downside to this?

Environment:
```
OS: Linux
pytorch_lightning version: 1.5.10
torch version: '1.11.0.post2'
```

**Comment by rohitgr7** [comment]
> I recently noticed that .load_from_checkpoint() has some unintuitive features, for example not setting the correct model.current_epoch (https://github.com/Lightning-AI/lightning/issues/12819) and needing users to manually save the hyperparameters.

it doesn't load the current_epoch as explained in the issue but it does load the hyperparameters if they are saved using `self.save_hyperparameters()`.

> I also noticed that if I just use the self.trainer.predict api, then self.trainer.model is automatically loaded with the correct weights and current_epoch

because model doesn't store the current epoch, it refers the current_epoch from Trainer and since you are using the same trainer instance here, the state persists.
